File: PharmacySrenity.Web/Modules/Product/Product/ProductColumns.cs
using PharmacySrenity.Modules.Product.Product;

namespace PharmacySrenity.Product.Columns;

[ColumnsScript("Product.Product")]
[BasedOnRow(typeof(ProductRow), CheckNames = true)]
public class ProductColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    
    public decimal Price { get; set; }
    public string Description { get; set; }
    [QuickFilter]
    [Hidden]
    public Guid? CategoryId { get; set; }
    public string CategoryName { get; set; }
    [QuickFilter]
    public Gender? Gender { get; set; }
}

File: PharmacySrenity.Web/Modules/ServerTypes/Product/ProductColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Gender } from "../Modules/Product.Product.Gender";
import { ProductRow } from "./ProductRow";

export interface ProductColumns {
    Id: Column<ProductRow>;
    Name: Column<ProductRow>;
    Price: Column<ProductRow>;
    Description: Column<ProductRow>;
    CategoryId: Column<ProductRow>;
    CategoryName: Column<ProductRow>;
    Gender: Column<ProductRow>;
}

export class ProductColumns extends ColumnsBase<ProductRow> {
    static readonly columnsKey = 'Product.Product';
    static readonly Fields = fieldsProxy<ProductColumns>();
}

[Gender]; // referenced types